packages/server/api/src/app/ee/platform/platform-plan/stripe-helper.ts

```diff
@@ -44,6 +44,95 @@ export const stripeHelper = (log: FastifyBaseLogger) => ({
+
+        const lineItems: Stripe.Checkout.SessionCreateParams.LineItem[] = [
+            {
+                price: AI_CREDIT_PRICE_ID,
+            },
+        ]
+
+        if (!isNil(extraActiveFlows) && extraActiveFlows > 0) {
+            lineItems.push({
+                price: ACTIVE_FLOW_PRICE_ID,
+                quantity: extraActiveFlows,
+            })
+        }
+
+        const session = await stripe.checkout.sessions.create({
+            payment_method_types: ['card'],
+            line_items: lineItems,
+            mode: 'subscription',
+            subscription_data: {
+                metadata: {
+                    platformId,
+                },
+            },
+            allow_promotion_codes: true,
+            customer: customerId,
+            success_url: `${frontendUrl}/platform/setup/billing/success?action=create`,
+            cancel_url: `${frontendUrl}/platform/setup/billing/error`,
+        })
+        
+        return session.url!
+    },
+    handleSubscriptionUpdate: async (params: HandleSubscriptionUpdateParams): Promise<string> => {
+        const { extraActiveFlows, isUpgrade, subscriptionId, isFreeDowngrade } = params
+
+        const stripe = stripeHelper(log).getStripe()
+        assertNotNullOrUndefined(stripe, 'Stripe is not configured')
+
+        try {
+            const subscription = await stripe.subscriptions.retrieve(subscriptionId, {
+                expand: ['items.data.price'],
+            })
+
+            const schedules = await stripe.subscriptionSchedules.list({
+                customer: subscription.customer as string,
+                limit: 10,
+            })
+        
+            const relevantSchedules = schedules.data.filter(schedule => 
+                schedule.subscription === subscription.id || 
+            schedule.status === 'active' || 
+            schedule.status === 'not_started',
+            )
+
+            if (isUpgrade) {
+                for (const schedule of relevantSchedules) {
+                    await stripe.subscriptionSchedules.release(schedule.id)
+                }
+
+                await updateSubscription({ stripe, subscription, extraActiveFlows })
+            }
+            else {
+                if (relevantSchedules.length > 0) {
+                    const schedule = relevantSchedules[0]
+                    await updateSubscriptionSchedule({ stripe, scheduleId: schedule.id, subscription, logger: log, extraActiveFlows, isFreeDowngrade })
+                
+                    for (let i = 1; i < relevantSchedules.length; i++) {
+                        await stripe.subscriptionSchedules.release(relevantSchedules[i].id)
+                    }
+                }
+                else {
+                    await createSubscriptionSchedule({ stripe, subscription, logger: log, extraActiveFlows, isFreeDowngrade })
+                }
+            }
+            return `/platform/setup/billing/success?action=${isUpgrade ? 'upgrade' : 'downgrade'}`
```

Comment:
Nit: the backend should not be aware from the frontend routes, It should return just success / error / status what ever the raw information the redirect should be handled in the frontend.